src/compiler/scala/tools/nsc/typechecker/Implicits.scala

```diff
@@ -995,6 +996,82 @@ trait Implicits {
         // most frequent one first
         matches sortBy (x => if (isView) -x.useCountView else -x.useCountArg)
       }
+
+      /** Sorted list of eligible implicits.
+       */
+      private def eligibleNew = {
+        final case class Candidate(info: ImplicitInfo, level: Int)
```

Comment:
- As a farther step to remove extra allocations, we can _unzip_ the `matches` array into an `Array[ImplicitInfo]` and an `Array[Int]`, and thus get rid of the `Candidate` class.
- Going farther, and assuming that 255 levels should be enough for anyone, perhaps we could make `level` into an `Array[Byte]` ?